tests/integration/pilot/workloadentry_test.go

```diff
@@ -0,0 +1,141 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilot
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"testing"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/check"
+	"istio.io/istio/pkg/test/framework/components/echo/common/deployment"
+	"istio.io/istio/pkg/test/framework/resource/config/apply"
+)
+
+func TestWorkloadEntry(t *testing.T) {
+	// nolint: staticcheck
+	framework.NewTest(t).
+		Features("traffic.reachability").
+		Run(func(t framework.TestContext) {
+			// Define a simulated EW gateway
+			gatewayCfg := `apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: ingress-ew
+  namespace: istio-system
+spec:
+  selector:
+    app: istio-ingressgateway
+  servers:
+  - port:
+      number: 8443
+      name: https
+      protocol: TLS
+    hosts:
+    - serviceentry.mesh.cluster.local
+    tls:
+      mode: AUTO_PASSTHROUGH
+`
+			// Setup a Gateway to act as simulated EW
+			if err := t.ConfigIstio().YAML("istio-system", gatewayCfg).Apply(apply.CleanupConditionally); err != nil {
+				t.Fatal(err)
+			}
+
+			// get pod IP of simulated EW gateway so we can use that IP in as the WorkloadEntry address
```

Comment:
Do you really want the pod IP? Not the Service IP?

you likely want `ist.EastWestGatewayFor(cfg.Cluster).AddressForPort(15443)`.